integreat_cms/cms/models/abstract_content_translation.py

```diff
@@ -579,6 +583,68 @@ def save(self, *args: Any, **kwargs: Any) -> None:
             self.last_updated = timezone.now()
         super().save(*args, **kwargs)
 
+    @transaction.atomic
+    def cleanup_autosaves(self) -> None:
+        """
+        Delete all but the last ``keep`` AUTO_SAVEs
+        and renumber all affected versions to be continuous.
+
+        """
+        logger.debug("Cleaning up old autosaves")
+        auto_saves = list(
+            self.foreign_object.translations.filter(
+                language=self.language, status=status.AUTO_SAVE
+            )
+        )
+
+        if (last_version := self.latest_version) is None or not auto_saves:
+            return
+
+        last_version = last_version.version
+        previous_version = None
+        # pylint:disable=undefined-loop-variable
+        for index, auto_save in enumerate(auto_saves):
+            if (
+                previous_version is not None
+                and auto_save.version < previous_version - 1
+                or auto_save.version < last_version - 1
+            ):
+                index -= 1  # noqa: PLW2901
+                break
+            previous_version = auto_save.version
+        delete_auto_saves = auto_saves[index + 1 :]
```

Comment:
If I need 5 minutes to understand 10 lines of code, then it's not good code.
Please reconsider if you really need this complex loop, and only if you really need it, write a few lines of comments what you are doing here and why you are doing it.
If I understand correctly, you are trying to find the first autosave which is not the immediate predecessor of the following autosave-chain?
I guess the much simpler solution would be to query the latest version which is not an autosave (or the second latest version, I didn't fully understand which autosaves we're keeping now) and just do a simple filter:
```
last_manual_save = self.foreign_object.translations.filter(
    language=self.language
).exclude(status=status.AUTO_SAVE).first()

delete_auto_saves = self.foreign_object.translations.filter(
    language=self.language,
    status=status.AUTO_SAVE,
    version__lt=last_manual_save.version,
)
```